api/upload.py

```diff
@@ -174,20 +209,36 @@ def upload(self, strategy):
             if not user:
                 self.abort(403, 'Uploading requires login')
 
+        if strategy in ['label', 'uid', 'uid-match', 'reaper']:
+            strategy = strategy.replace('-', '')
+            strategy = getattr(Strategy, strategy)
+        else:
+            self.abort(500, 'strategy {} not implemented'.format(strategy))
```

Comment:
I don't think we need the 500 here because something not accounted for won't make it past the routing logic in `api.py`.